src/rules/newlinePerChainedCallRule.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isCallExpression, isIdentifier, isPropertyAccessExpression } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "..";
+
+interface Options {
+    maxChainLength: number;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "newline-per-chained-call",
+        description: Lint.Utils.dedent`
+            Requires that chained property accessors be broken apart onto separate lines.`,
```

Comment:
`chained property accessors` or `chained method calls`?

Reply:
also note that the corresponding eslint rule only counts method calls and not property accesses

Reply:
So method calls on their own line if the chain length--including prop accessors--is greater than n? Or should chain length include method calls only?

Reply:
I guess method calls only is the best to start with. Otherwise you'll get a lot more errors with little benefit.
For example `chai` assertions:
```ts
expect({a: 1}).to.not.have.property('b');
```

Reply:
Good point. Your single-line `chai` example looks fine to me, but if it had a third call I'd want each call to be on its own line:
```
expect({ a: 1 })
    .to.not.have
    .property('b').more.accessors // Should this be okay?
    .methodCall(1);
```

Reply:
And this makes me realize that my scheme of counting `\n` chars is not very flexible...

Reply:
I guess it's best to check if there is a newline after the method call. No counting at all.
```ts
// allowing
expect({ a: 1 })
    .to.not.have.property('b');
// disallowing
expect({ a: 1 }).to.not.have
    .property('b')
```
The only exception is the first method call:
```ts
// not allowed
foo.bar()
    .baz();
// same with
this.bar().
    .baz();

// allowed
foo
    .bar()
    .baz();
```

Reply:
Okay nice, that's a much better solution. I have only one thing I'd change in your examples, and it's purely stylistic. I think property accessors between method calls look better like this:
```
expect({ a: 1 })
    .to.not.have
    .property('b');
```
... the condition being that prop accessors between method calls cannot be on the same line as either method call. I think? It may be too convoluted for a config option. I think this is the behavior I would want in the default setting, though.